projects/step-core/src/lib/modules/table/services/items-per-page.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { UserService } from '../../../client/generated';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ItemsPerPageService {
+  private defaultItemsPerPage: Array<number> = [10, 25, 50, 100];
+  private customItemsPerPage?: number | boolean;
+
+  constructor(private _userService: UserService) {}
+
+  public getItemsPerPage(loadedUserPreferences: Function): any {
+    if (this.customItemsPerPage === undefined) {
+      this._userService.getPreferences().subscribe((preferences) => {
+        if (preferences?.preferences?.['tables_itemsperpage']) {
+          this.customItemsPerPage = Number(preferences.preferences['tables_itemsperpage']);
+          loadedUserPreferences(Number(preferences.preferences['tables_itemsperpage']));
+        }
+      });
+    }
+    const itemsPerPage = this.defaultItemsPerPage;
```

Comment:
you are assigning ```itemsPerPage``` without changing the reference (```this.defaultItemsPerPage.slice()``` or ```[...defaultItemsPerPage]```) so that later when you do ```.unshift(this.customItemsPerPage)``` it would also mutate ```this.defaultItemsPerPage```

Reply:
```suggestion
    const itemsPerPage = [...this.defaultItemsPerPage];
```